src/workshopselector/WorkshopSelector.java

```diff
@@ -6,65 +6,80 @@
  */
 package workshopselector;
 
-import java.awt.Color;
-import java.awt.FlowLayout;
 import javax.swing.*;
-import javax.swing.JFrame;
-import javax.swing.JLabel;
-import javax.swing.JList;
-import javax.swing.JScrollPane;
-import javax.swing.ListSelectionModel;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
 
 
- 
-public class WorkshopSelector {
+public class WorkshopSelector implements ActionListener {
+
+    private JList listWorkshops;
+    private JList listLocation;
+    private JLabel totalCostLabel;
+    private int[] allLocationFees;
+    private int[] allWorkshopFees;
+    private int[] allWorkshopDurations;
 
     /**
      * @param args the command line arguments
      */
     public static void main(String[] args) {
+        new WorkshopSelector().build();
```

Comment:
I moved all the code out of this static method and into instance methods so that variables could be properly shared.